lib/hammer_cli_katello/repository.rb

```diff
@@ -292,14 +292,19 @@ def request_params
       build_options :without => %w(uuids ids)
     end
 
-    class ExportCommand < HammerCLIKatello::Command
-      resource :repositories, :export
+    class ExportCommand < HammerCLIKatello::SingleResourceCommand
+      include HammerCLIForemanTasks::Async
+      include RepositoryScopedToProduct
+
+      action :export
       command_name "export"
 
-      success_message _("Exported repository")
-      failure_message _("Could not export repository")
+      success_message _("Repository is being exported in task %{id}")
+      failure_message _("Could not export the repository")
```

Comment:
:+1: had to test this by piping to less, `include HammerCLIForemanTasks::Async` outsmarted me and knew I was on an interactive terminal so I got a progress bar otherwise.